forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -116,6 +116,85 @@ pub fn is_car_or_tmp(path: &Path) -> bool {
     ext == "car" || ext == "tmp" || ext == "aria2"
 }
 
+/// gets the size of a snapshot from the given provider
+pub async fn snapshot_fetch_size(
+    config: &Config,
+    provider: &Option<SnapshotServer>,
+) -> anyhow::Result<u64> {
+    let server = match provider {
+        Some(s) => s,
+        None => match config.chain.name.to_lowercase().as_str() {
+            "mainnet" => &SnapshotServer::Filecoin,
+            "calibnet" => &SnapshotServer::Forest,
+            _ => anyhow::bail!("Fetch not supported for chain {}", config.chain.name),
+        },
+    };
+    match server {
+        SnapshotServer::Forest => {
+            let snapshot_fetch_config = match config.chain.name.to_lowercase().as_str() {
+                "mainnet" => bail!(
+                    "Mainnet snapshot fetch service not provided by Forest yet. Suggestion: use `--provider=filecoin` to fetch from Filecoin server."
+                ),
+                "calibnet" => &config.snapshot_fetch.forest.calibnet,
+                _ => bail!("Fetch not supported for chain {}", config.chain.name,),
+            };
+            let name = &snapshot_fetch_config.bucket_name;
+            let region = &snapshot_fetch_config.region;
+            let bucket = Bucket::new_public(name, region.parse()?)?;
+
+            // Grab contents of the bucket
+            let bucket_contents =
+                bucket.list(snapshot_fetch_config.path.clone(), Some("/".to_string()))?;
```

Comment:
Do we really need the delimiter as second param?

Reply:
I just copied over the code from the other sections 😆